fvm/state/accounts.go

```diff
@@ -32,17 +32,38 @@ func keyPublicKey(index uint64) string {
 	return fmt.Sprintf("public_key_%d", index)
 }
 
-type Accounts struct {
+type Accounts interface {
+	Exists(address flow.Address) (bool, error)
+	GetPublicKeyCount(address flow.Address) (uint64, error)
+	AppendPublicKey(address flow.Address, key flow.AccountPublicKey) error
+	GetPublicKey(address flow.Address, keyIndex uint64) (flow.AccountPublicKey, error)
+	SetPublicKey(address flow.Address, keyIndex uint64, publicKey flow.AccountPublicKey) ([]byte, error)
+	GetContractNames(address flow.Address) ([]string, error)
+	GetContract(contractName string, address flow.Address) ([]byte, error)
+	SetContract(contractName string, address flow.Address, contract []byte) error
+	DeleteContract(contractName string, address flow.Address) error
+	Create(publicKeys []flow.AccountPublicKey, newAddress flow.Address) error
+	GetValue(address flow.Address, key string) (flow.RegisterValue, error)
+	CheckAccountNotFrozen(address flow.Address) error
+	GetStorageUsed(address flow.Address) (uint64, error)
+	SetValue(address flow.Address, key string, value []byte) error
+	AllocateStorageIndex(address flow.Address) (uint64, error)
+	SetAccountFrozen(address flow.Address, frozen bool) error
+}
+
+var _ Accounts = &AccountsState{}
+
+type AccountsState struct {
```

Comment:
Maybe another name like StatefulAccounts

Reply:
Yep, I like it